File: skylyt-frontend/src/utils/sanitize.ts
/**
 * Sanitization utilities to prevent XSS and injection attacks
 */

/**
 * Sanitize string for safe logging to prevent log injection
 */
export const sanitizeForLogging = (input: any): string => {
  if (typeof input !== 'string') {
    input = String(input);
  }
  
  // Remove newlines, carriage returns, and other control characters
  return input
    .replace(/[\r\n\t\f\v]/g, ' ')
    .replace(/[\x00-\x1f\x7f-\x9f]/g, '')
    .trim();
};

/**
 * Sanitize HTML content to prevent XSS
 */
export const sanitizeHtml = (input: string): string => {
  if (typeof input !== 'string') return '';
  
  return input
    .replace(/[<>\"'&]/g, (match) => {
      const entities: Record<string, string> = {
        '<': '&lt;',
        '>': '&gt;',
        '"': '&quot;',
        "'": '&#x27;',
        '&': '&amp;'
      };
      return entities[match] || match;
    });
};

/**
 * Sanitize input for safe JSON usage to prevent NoSQL injection
 */
export const sanitizeForJson = (input: any): any => {
  if (typeof input === 'string') {
    // Remove potential JSON injection characters
    return input.replace(/[\{\}\[\]"'\\]/g, '');
  }
  
  if (typeof input === 'object' && input !== null) {
    const sanitized: any = Array.isArray(input) ? [] : {};
    for (const key in input) {
      if (input.hasOwnProperty(key)) {
        sanitized[sanitizeForJson(key)] = sanitizeForJson(input[key]);
      }
    }
    return sanitized;
  }
  
  return input;
};

/**
 * Validate and sanitize user input
 */
export const validateAndSanitize = (input: string, maxLength: number = 1000): string => {
  if (typeof input !== 'string') return '';
  
  return sanitizeHtml(input.slice(0, maxLength));
};

File: skylyt-frontend/skylyt-travelhub-backend/app/utils/sanitize.py
import re
import html


def sanitize_for_logging(text: str) -> str:
    """Sanitize text for safe logging by removing newlines and control characters"""
    if not text:
        return ""
    
    # Remove newlines and carriage returns
    sanitized = re.sub(r'[\r\n]', '', str(text))
    
    # Remove other control characters except tab
    sanitized = re.sub(r'[\x00-\x08\x0B\x0C\x0E-\x1F\x7F]', '', sanitized)
    
    # Truncate if too long
    if len(sanitized) > 100:
        sanitized = sanitized[:97] + "..."
    
    return sanitized


def sanitize_html(text: str) -> str:
    """Sanitize HTML content"""
    if not text:
        return ""
    
    return html.escape(str(text))